### PR TITLE
Team 1 Conceptual Design

### PR DESCRIPTION
**Pull request by DJ-Moye**
Questions for Micah and known issues:

Questions
1. Should each subsystem add to High-Level Solution, or should it be one overarching solution?
2. Should each subsystem include its own part of the block diagram that it should then elaborate on, or should it be covered as a whole in High-Level Solution. 

Known Issues
1. Specifications and Constraints not elaborated on in first section.
2. Gantt Chart for Timeline missing.
3. Missing specifications for General Sensors and Navigation and Master Control.
4. Block Diagram missing sections for Camera and Power Management.

### CODE REVIEW

File: Reports/Conceptual-Design.md [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
... a clearly labeled emergency stop button according to Specification 7. If there are multiple independently acting ‘Robot Units’, each shall have their own emergency stop button. Regarding power system requirements, the robot shall not go beyond 30V, as given by Specification 12. It shall also implement proper circuit shielding and isolation based on the disturbances from background interference in the competition environment, as given by Specification 11. Constraint 3 deals with International Electrotechnical Commision regulations for power supply, emergency stop, and circuit protection requirements. From this standard, the circuitry of the robot shall account for and add appropriate devices to promote safety measures such as overcurrent, overload, and over-temperature protection [3]. The power subsystem of this robot will be the electrical backbone of the project. Without a structured model of power management and distribution, the robot would not be able to meet all functions, specifications and constraints. For this subsystem to work correctly with the scope of the project, we must install a battery/(ies) so that the robot can be powered without a cord attached to a wall socket.
```
Comment (micahr234) [inline Reports/Conceptual-Design.md:260]: List "SHALL" statements.

File: Reports/Conceptual-Design.md [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+
```
Comment (micahr234) [inline Reports/Conceptual-Design.md:261]: Be specific. What voltages are needed? What current is required? For how long? How much energy will be needed?

File: Reports/Conceptual-Design.md [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+The infrared laser projector is safe for use in the vicinity of humans. 
+
+Resources and materials uses for this subsystem will be built following an expected use.
+This use will be listed in their datasheets, and their datasheets will be followed properly.
+The rules point out safety measures that will be taken throughout the competition.
+A rule that sensors need to abide by is no visible strobe lights.
+Furthermore, if using non-visble lasers, lasers will need to be safe for human interaction.
+
+## Timeline
+## Statement of Contributions
```
Comment (micahr234) [inline Reports/Conceptual-Design.md:300]: I need to know who will be responsible for the detailed design of each subsystem.

File: Reports/Conceptual-Design.md [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+The reflectance array, inertial measurement unit, and LiDAR will take the most work.
+The other two should be a simple voltage comparison, or trigger.
+The first three will need to be tested repeatedly.
+The gyroscope on the inertial measurement unit will need to be tested with the motors running and with the magnetic material.
+
+
+### Navigation and Master Control
+
+#### Atomic Subsystem Specifications
+![Atomic Hardware Block Diagram](https://github.com/ACruz-42/F24_Team1_CapstoneDemo/blob/d2995a028545a98730a56ce987d4fb0d5fdcdbad/Reports/Photos/Conceptual%20Design/Master_Control_Subsystem_V1.png)
```
[the remark this answers is not in the document]
Comment (ACruz-42) [reply Reports/Conceptual-Design.md]: Removed.

File: Reports/Conceptual-Design.md (hunk 3 above)
Comment (ACruz-42) [reply Reports/Conceptual-Design.md:340]: Added below each subsystem's section header

File: Reports/Conceptual-Design.md [hunk 5]
```diff
@@ -0,0 +1,331 @@
+﻿# Conceptual Design
+## Introduction
+## Restate Shall/May Statements
```
[the remark this answers is not in the document]
Comment (ACruz-42) [reply Reports/Conceptual-Design.md:10]: Added

File: Reports/Conceptual-Design.md [hunk 6]
```diff
@@ -0,0 +1,331 @@
+﻿# Conceptual Design
+## Introduction
```
[the remark this answers is not in the document]
Comment (ACruz-42) [reply Reports/Conceptual-Design.md:2]: Added

File: Reports/Conceptual-Design.md [hunk 7]
```diff
@@ -0,0 +1,331 @@
...
+
+A camera is needed for specifcations 1, 2-vii, 2-viii, and 4 and constraint 1.
+By having a camera, a robot can see very similarly to a human, but needs to have that information filter and processed in a way the robot can understand.
+If it understands what it can effectively see, it shall be able to act autonomously, meeting specification 1.
+Specifcations 2-vii and 2-viii both shall have the robot have a way to identify the materials in the arena.
+Specifcation 4 shall have the robot detect different objects and obstacles.
+The camera will be the only sensor in the front getting active feedback, due to the constraints placed by the device used to collect the materials.
+The camera shall help keep the robot from running into obstacles in the front. 
+Constraint 1 shall have the robot read April Tags. 
+A camera is needed to read these.
```
[the remark this answers is not in the document]
Comment (DJ-Moye) [reply Reports/Conceptual-Design.md]: List added and written explanation removed

File: Reports/Conceptual-Design.md [hunk 8]
```diff
@@ -0,0 +1,331 @@
...
+
+Specifications 2-vii and 2-viii require the material to be in its proper container for maximum points.
+The easiest way to distinguish between the two is to use a magnetic, or hall effect, sensor.
+Hall effect sensors produce a more positive or negative voltage depending on whether it detects a positive or negative magnetic field.
+A different voltage gives a way for the microcontroller to determine where a certain material should go [11].
+
+A second option is to use weight.
+This require whatever is being used to transport the material to be weighed as well.
+Unless, the material is placed directly on a scale, but then it also has to be moved off of the scale.
+The weight sensor would also need to account for the acceleration of the robot, which gives more room for error and causes more work for software.
```
[the remark this answers is not in the document]
Comment (DJ-Moye) [reply Reports/Conceptual-Design.md]: Final chosen sensor added. Source added for the reflectance array.

File: Reports/Conceptual-Design.md [hunk 9]
```diff
@@ -0,0 +1,331 @@
...
+
+The robot movement will need to comply with Game Manual specifications. There are specific rules in place for the safety of the team and the referees/technicians at the competition. Specification 10 details the need for the robot to cease operation of all units after the given 3 minutes of play. Therefore, the motor control programming and mechanisms will need to have measures in place to quickly and safely halt the robot and any actions it is performing toward the end of the time of play.
```
[the remark this answers is not in the document]
Comment (SeanBorchers) [reply Reports/Conceptual-Design.md]: Removed second paragraph that was focused more on rule specifications than ethical considerations.

File: Reports/Conceptual-Design.md (hunk 1 above)
Comment (SeanBorchers) [reply Reports/Conceptual-Design.md]: Format changed to listed SHALL statements using the given specifications and constraints.

File: Reports/Conceptual-Design.md (hunk 2 above)
Comment (SeanBorchers) [reply Reports/Conceptual-Design.md:336]: Added specific voltage and current values for power requirements. Also discussed the time frame for when the components using power would be operational.

File: Reports/Conceptual-Design.md [hunk 10]
```diff
@@ -0,0 +1,331 @@
...
+The datasheet for the camera will be followed and the intended use matches closely with the designed use.
+The infrared laser projector is safe for use in the vicinity of humans. 
+
+Resources and materials uses for this subsystem will be built following an expected use.
+This use will be listed in their datasheets, and their datasheets will be followed properly.
+The rules point out safety measures that will be taken throughout the competition.
+A rule that sensors need to abide by is no visible strobe lights.
+Furthermore, if using non-visble lasers, lasers will need to be safe for human interaction.
+
+## Timeline
```
[the remark this answers is not in the document]
Comment (SeanBorchers) [reply Reports/Conceptual-Design.md:412]: Added

File: Reports/Conceptual-Design.md [hunk 11]
```diff
@@ -0,0 +1,331 @@
...
+1)	The robot shall sustain power of 30V maximum. Rule R16 [1].
+	
+### Constraints
+1)	The robot shall read April Tags to help with Specifications 1, 2-v, 2-vii, 2-viii, and 2-ix. 
+1)	The robot shall have a user manual that explains functionality and design intent for each subsystem.
+1)	The robot shall adhere to applicable requirements in standard IEC 60204-1 pertaining to electrical supply, electromagnetic compatibility, emergency stop, and control circuit protection. The power-based and electromagnetic interactions of components in the motor control subsystem will need to adhere to this standard [3].
+
+## High-Level Solution
+
+### Motor Control
```
[the remark this answers is not in the document]
Comment (SeanBorchers) [reply Reports/Conceptual-Design.md:66]: A high level overview of each of the subsystems and mechanisms of the design solution has been included.

File: Reports/Conceptual-Design.md [hunk 12]
```diff
@@ -0,0 +1,331 @@
...
+
+
+### Hardware Block Diagram
+![Hardware Block Diagram](https://github.com/ACruz-42/F24_Team1_CapstoneDemo/blob/1fa993cd7f82cd33691eebe0be5f6cf70c7abc0e/Reports/Photos/Conceptual%20Design/BlockDiagram.jpg)
```
[the remark this answers is not in the document]
Comment (SeanBorchers) [reply Reports/Conceptual-Design.md]: Added

File: Reports/Conceptual-Design.md [hunk 13]
```diff
@@ -0,0 +1,331 @@
...
+The first three will need to be tested repeatedly.
+The gyroscope on the inertial measurement unit will need to be tested with the motors running and with the magnetic material.
+
+
+### Navigation and Master Control
+
+#### Atomic Subsystem Specifications
+![Atomic Hardware Block Diagram](https://github.com/ACruz-42/F24_Team1_CapstoneDemo/blob/d2995a028545a98730a56ce987d4fb0d5fdcdbad/Reports/Photos/Conceptual%20Design/Master_Control_Subsystem_V1.png)
+The main component in the navigation and master control subsystem is the Jetson Nano. The choice of the Jetson Nano is discussed below. The Jetson Nano will take in data from the sensor-related subsystems, interpret that data, form , utilize a navigation algorithm to plot a course through the game field maximizing the points obtained, then communicate to the motor subsystem the distance and angle to the desired position of the robot. Additionally, at the start of each game round, the Jetson Nano will start and maintain the movement of the auxiliary motors (such as for the auger, the roller, and potentially the docking of the cosmic shipping containers).
+#### Comparative Analysis of Potential Solutions.
```
[the remark this answers is not in the document]
Comment (SeanBorchers) [reply Reports/Conceptual-Design.md:274]: The format was changed to list a series of SHALL statements based on the applicable specifications.

File: Reports/Conceptual-Design.md [hunk 14]
```diff
@@ -0,0 +1,331 @@
...
+The robot will need sensors to detect when the start LED turns on, specifications 2 i and ii.
+The robot will need a sensor capable of detecting magnetic fields for the purpose of sorting, specifications 2 vii and viii. 
+The robot cannot make any choices without input from the outside world, this input is obtained through sensors.
+The robot will need to make many choices about movement and navigation, to asist in making these choices the robot will need information about its current position, such as its relation to the game field walls and cave walls, lines on the game field, as well as its relation to its own position at an earlier point in time, it will also need to be able to detect magnetic fields to sort the astral material, and when a start LED comes on to signal the beginning of a match.
+Specifications 2 iii, v, and 4, 5 will require the robot to have information about its position.
+Specifications 2 vii, and viii will require the robot to have the ability to detect magnetic fields.
+Finally specifications 2 i and ii will require the robot to be able to detect a start LED.
```
[the remark this answers is not in the document]
Comment (SeanBorchers) [reply Reports/Conceptual-Design.md]: Modified to include SHALL statements.

File: Reports/Conceptual-Design.md [hunk 15]
```diff
@@ -0,0 +1,331 @@
...
+
+The robot movement will need to comply with Game Manual specifications. There are specific rules in place for the safety of the team and the referees/technicians at the competition. Specification 10 details the need for the robot to cease operation of all units after the given 3 minutes of play. Therefore, the motor control programming and mechanisms will need to have measures in place to quickly and safely halt the robot and any actions it is performing toward the end of the time of play. 
+
+The Game Manual provided structures for the competition to follow IEEE guidelines and general electrical standards. Resources mentioned such as datasheets and procedures correlated with clean power transmission will be followed closely. For example, the E-stop mentioned in the rules are for the safety of the competitors, judges, and audience in the case of a motor malfunction.
+
+The datasheet for the camera will be followed and the intended use matches closely with the designed use.
+The infrared laser projector is safe for use in the vicinity of humans. 
+
+Resources and materials uses for this subsystem will be built following an expected use.
```
[the remark this answers is not in the document]
Comment (SeanBorchers) [reply Reports/Conceptual-Design.md:406]: Rewritten for clarity.

File: Reports/Conceptual-Design.md [hunk 16]
```diff
@@ -0,0 +1,331 @@
...
+
+#### Comparative Analysis of Potential Solutions
+
+The robot will need a camera with both RGB and dpeth sensing capabillites.
+The camera has 2 cameras in it, 1 is a 2D camera which will be used like a normal 2D camera for video and RGB values.
+The second camera is an infrared camera which uses an infrared laser detector which seperates objects into layers. 
+The primary purpose of the camera will be to identify the astral material and then accurately place it on the game field for the navigation algorithm.
+The camera will use the 2D RGB camera to find and differentiate the different objects, including astral material, game field walls, and cave walls.
+It will then use the infrared camera to accurately place the objects in relation to the robot.[24]
+
```
[the remark this answers is not in the document]
Comment (SeanBorchers) [reply Reports/Conceptual-Design.md:106]: Added. The camera will interface with the microcontroller using a USB connection.

**Review by micahr234 — APPROVED** [review]

**Review by srizvi-tntech — APPROVED** [review]

**Pull request merged**